tests/unit/test_sync_process.py

```diff
@@ -0,0 +1,180 @@
+import queue
+import time
+from typing import List
+from unittest.mock import Mock
+
+import pytest
+from neptune_api.proto.neptune_pb.ingest.v1.common_pb2 import (
+    UpdateRunSnapshot,
+    Value,
+)
+from neptune_api.proto.neptune_pb.ingest.v1.pub.client_pb2 import SubmitResponse
+from neptune_api.proto.neptune_pb.ingest.v1.pub.ingest_pb2 import RunOperation
+
+from neptune_scale.exceptions import NeptuneSynchronizationStopped
+from neptune_scale.sync.queue_element import (
+    BatchedOperations,
+    SingleOperation,
+)
+from neptune_scale.sync.sync_process import SenderThread
+from neptune_scale.util.shared_var import SharedInt
+
+
+def response(request_ids: List[str], status_code: int = 200):
+    body = SubmitResponse(request_ids=request_ids, request_id=request_ids[-1] if request_ids else None)
+    content = body.SerializeToString()
+    return Mock(status_code=status_code, content=content, parsed=body)
+
+
+def single_operation(update: UpdateRunSnapshot, sequence_id):
+    operation = RunOperation(update=update)
+    return SingleOperation(
+        sequence_id=sequence_id,
+        timestamp=time.process_time(),
+        operation=operation.SerializeToString(),
+        is_batchable=True,
+        metadata_size=update.ByteSize(),
+        batch_key=None,
+    )
+
+
+def test_sender_thread_ends_when_queue_empty():
```

Comment:
It should be fine, because I don't call thread.start() in tests, I only call work() directly. 
This way work() should get executed in the main thread running the test, no new thread should be started at all.

Reply:
hmm, maybe it's pointless to test it at all, because it's a detail whether work ends and is looped in run() or the loop is inside work()...

but test_sender_thread_work_finishes_when_queue_empty should be more correct